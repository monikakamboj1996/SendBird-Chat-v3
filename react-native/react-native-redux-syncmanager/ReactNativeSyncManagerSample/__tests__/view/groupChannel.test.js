import 'react-native';
import React from 'react';
import GroupChannelView from '../../src/view/groupChannel';
import renderer from 'react-test-renderer';

describe('view/groupChannel', () => {
  it('renders correctly user message', () => {
    const message = {
      isFileMessage: () => false,
      isUserMessage: () => true,
      url: 'http://some url',
      createdAt: Date.now(),
      updatedAt: Date.now(),
      coverUrl: 'coverUrl',
      unreadMessageCount: 20,
      message: 'text message'
    };
    const channel = {
      lastMessage: message,
      members: [
        {
          nickname: 'nick1'
        },
        {
          nickname: 'nick2'
        }
      ]
    };
    const tree = renderer.create(<GroupChannelView channel={channel} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly file message', () => {
    const message = {
      isFileMessage: () => true,
      isUserMessage: () => true,
      url: 'http://some url',
      createdAt: Date.now(),
      updatedAt: Date.now(),
      coverUrl: 'coverUrl',
      unreadMessageCount: 20,
      name: 'file message'
    };
    const channel = {
      lastMessage: message,
      members: [
        {
          nickname: 'nick1'
        },
        {
          nickname: 'nick2'
        }
      ]
    };
    const tree = renderer.create(<GroupChannelView channel={channel} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
