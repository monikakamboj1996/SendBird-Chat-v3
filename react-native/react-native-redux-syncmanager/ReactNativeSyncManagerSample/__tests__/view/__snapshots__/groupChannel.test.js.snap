// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view/groupChannel renders correctly file message 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#eee",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 17,
          "height": 34,
          "width": 34,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#bdbdbd",
            "borderRadius": 17,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoadEnd={[Function]}
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "flex": 1,
              "height": null,
              "width": null,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "flex": 1,
                "height": null,
                "justifyContent": "center",
                "width": null,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
      </View>
    </View>
    <View
      width={16}
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text>
            nick1, nick2
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#e3e3e3",
                "borderRadius": 4,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "textAlign": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#878D99",
                  "fontSize": 10,
                }
              }
            />
          </View>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#878D99",
                "fontSize": 10,
              }
            }
          >
            a few seconds ago
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#878D99",
              "fontSize": 12,
              "marginTop": 3,
            }
          }
        >
          file message
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`view/groupChannel renders correctly user message 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#eee",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 17,
          "height": 34,
          "width": 34,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#bdbdbd",
            "borderRadius": 17,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoadEnd={[Function]}
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "flex": 1,
              "height": null,
              "width": null,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "flex": 1,
                "height": null,
                "justifyContent": "center",
                "width": null,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
      </View>
    </View>
    <View
      width={16}
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text>
            nick1, nick2
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#e3e3e3",
                "borderRadius": 4,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "textAlign": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#878D99",
                  "fontSize": 10,
                }
              }
            />
          </View>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#878D99",
                "fontSize": 10,
              }
            }
          >
            a few seconds ago
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#878D99",
              "fontSize": 12,
              "marginTop": 3,
            }
          }
        >
          text message
        </Text>
      </View>
    </View>
  </View>
</View>
`;
